File: PDOO-Civitas/CivitasR/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Casilla_juez < Casilla
    
    attr_reader :nombre
    
    def initialize(nombre, num_casilla_carcel)
      
      super(nombre)
      @@carcel = num_casilla_carcel
      
    end
    
    def recibe_jugador(iactual, todos)
     
      if(jugador_correcto(iactual,todos))
        informe(iactual,todos)
        todos.at(iactual).encarcelar(@@carcel)
      end
    end
    
    public
    def to_s
      casilla="\n"
        if(@nombre!=nil)
          casilla = casilla + "\n  Nombre casilla: "+@nombre.to_s
        end
        if(@@carcel!=-1.0)
            casilla = casilla + "\n  Carcel: " + @@carcel.to_s
        end
        
        casilla = casilla +"\n  Tipo Casilla: JUEZ"
        
        return casilla
    end
  end
end

File: PDOO-Civitas/CivitasJ/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class CasillaJuez extends Casilla{
    
    private String nombre;
    private static int carcel;
    
    CasillaJuez(int numCasillaCarcel, String nombre){
        super(nombre);
        this.carcel = numCasillaCarcel;
        this.informacion = "Has caído en la casilla Juez. Buena Suerte.";
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
    
        if(this.jugadorCorrecto(iactual, todos)){
            this.informe(iactual, todos);
            todos.get(iactual).encarcelar(carcel);
        }
    }
    
    @Override
    public String toString(){
        
        String Casilla="\n";
        
        if(nombre!=null)
            Casilla = "\n  Nombre casilla: "+nombre;
        if(carcel!=-1.0)
            Casilla =Casilla + "\n  Carcel: " + carcel;
        
        Casilla = Casilla +"\n  Tipo Casilla: JUEZ";
        
        return Casilla;
    }
    
}
